packages/legend-extension-dsl-serializer/src/components/studio/SchemaSetElementEditor.tsx

```diff
@@ -0,0 +1,280 @@
+ * limitations under the License.
+ */
+
+import { useState } from 'react';
+import { observer } from 'mobx-react-lite';
+import { FaLock, FaPlus } from 'react-icons/fa';
+import {
+  CaretDownIcon,
+  clsx,
+  ContextMenu,
+  DropdownMenu,
+  MenuContent,
+  MenuContentItem,
+  ResizablePanel,
+  ResizablePanelGroup,
+  ResizablePanelSplitter,
+  ResizablePanelSplitterLine,
+} from '@finos/legend-art';
+import { useEditorStore } from '@finos/legend-studio';
+import type { SchemaSet } from '../../models/metamodels/pure/model/packageableElements/schemaSet/SchemaSet';
+import { Schema } from '../../models/metamodels/pure/model/packageableElements/schemaSet/Schema';
+import { SchemaSetEditorState } from '../../stores/studio/SchemaSetEditorState';
+import { EDITOR_LANGUAGE, TextInputEditor } from '@finos/legend-application';
+import type { editor } from 'monaco-editor';
+import { FORMAT_TYPE } from '../../models/metamodels/pure/model/packageableElements/schemaSet/SchemaSet';
+
+const SchemaSetFormatBasicEditor = observer(
+  (props: { schemaSet: SchemaSet; isReadOnly: boolean }) => {
+    const { schemaSet, isReadOnly } = props;
+    const changeType =
+      (val: FORMAT_TYPE): (() => void) =>
+      (): void => {
+        !isReadOnly && schemaSet.setFormat(val);
+      };
+    return (
+      <div className="schema-set-format-editor">
+        <DropdownMenu
+          disabled={isReadOnly}
+          content={
+            <MenuContent className="schema-set-format-editor__dropdown">
+              {Object.values(FORMAT_TYPE).map((format) => (
+                <MenuContentItem
+                  key={format}
+                  className="schema-set-format-editor__option"
+                  onClick={changeType(format)}
+                >
+                  {format}
+                </MenuContentItem>
+              ))}
+            </MenuContent>
+          }
+          menuProps={{
+            anchorOrigin: { vertical: 'bottom', horizontal: 'right' },
+            transformOrigin: { vertical: 'top', horizontal: 'right' },
+          }}
+        >
+          <div className="schema-set-format-editor__type">
+            <div className="schema-set-format-editor__type__label">
+              {schemaSet.format}
+            </div>
+            <div className="schema-set-format-editor__type__icon">
+              <CaretDownIcon />
+            </div>
+          </div>
+        </DropdownMenu>
+      </div>
+    );
+  },
+);
+
+const SchemaBasicEditor = observer(
+  (props: { schema: Schema; isReadOnly: boolean }) => {
+    const { schema, isReadOnly } = props;
+    const changeId: React.ChangeEventHandler<HTMLInputElement> = (event) =>
+      schema.setId(event.target.value);
+    const changeLocation: React.ChangeEventHandler<HTMLInputElement> = (
+      event,
+    ) => schema.setLocation(event.target.value);
+    const editorOptions: editor.IEditorOptions & editor.IGlobalEditorOptions = {
+      lineNumbers: 'off',
+      fontFamily: 'Arial',
```

Comment:
why do we use `Arial` instead of using the default config for `TextInputEditor`

Reply:
> why do we use `Arial` instead of using the default config for `TextInputEditor`. You should check out the API of `TextInputEditor` to see the options

Id, location uses different fontFamily from the default config for TextInputEditor. It was looking really weird. So I changed it.

Reply:
what do you mean by `location` uses a different font? Could you attach a screenshot so I could see how weird it looks?

Reply:
> what do you mean by `location` uses a different font? Could you attach a screenshot so I could see how weird it looks?

![pic3](https://user-images.githubusercontent.com/87973126/136264981-ed3924d4-d6f9-40b9-95b3-f7307e25d150.PNG)